### ISSUE TITLE
Libratbag or ratbagd should perform a range check on button_{g,s}et_special

### ISSUE THREAD

**Issue by Hjdskes** [libratbag/libratbag#281]
For some unknown reason my device now has a special button mapping with value 4294967295; this number crashes Piper because it's not recognized in the bindings so there is 1) no RatbagdButton.ACTION_SPECIAL_FOO and 2) no entry in the map that translates these values to their human-readable strings with that value as key (ratbagctl is broken for the same reason). Granted, I could add a try/catch block around the map, but this might be better solved here where this value should be detected in either libratbag's or ratbagd's setter and getter methods to set/get ACTION_SPECIAL_UNKNOWN instead on any unknown value.

**Comment by phomes** [comment]
4294967295 is what you get when you cast a normal int of -1 to an unsigned int.

RATBAG_BUTTON_ACTION_SPECIAL_INVALID is -1 in the enum used to store the action_specials. In ratbagds function [ratbagd_button_get_special](https://github.com/libratbag/libratbag/blob/master/ratbagd/ratbagd-button.c#L119) to export the value over dbus the value is specified as unsigned. The "u" specifies an unsigned int:
```
special = ratbag_button_get_special(button->lib_button);
return sd_bus_message_append(reply, "u", special);
```
If we do not want to change the data type on dbus, then you can adjust the value for ACTION_SPECIAL_INVALID in piper instead. You can simply change it to
`ACTION_SPECIAL_INVALID = (1 << 32) -1`

**Comment by Hjdskes** [comment]
Changing the value for `ACTION_SPECIAL_INVALID` is another workaround, likely better than using a try/catch block. The best solution (imo) would be to change the interface or the internal value for `ACTION_SPECIAL_INVALID`, while we still can (the interface isn't declared stable yet, is it?). This way, all clients are automagically fixed.

**Comment by thparamboll** [comment]
@whot should we change the interface to signed int or deal with it in piper?

**Comment by phomes** [comment]
@whot should we change the interface to signed int or deal with it in piper?

(apologies for the spam. I accidentally sent the other comment from my work account )

**Comment by whot** [comment]
probably change it all to signed.  `ratbag_button_action_special` could change `INVALID` to 0 and still have everything work fine (after the required bumps everywhere). but for `ratbag_macro_event_type` we can't do that, we definitely need a `NONE` value there.

So bumping all the enums to signed integers instead seems like the best option here, it gives us the negative range for errors. And I'm struggling to come up with a sane universe where 2 billion enum values aren't enough to configure button macros.

**Comment by whot** [comment]
I **think** this one got effectively fixed as part of #516
